prow/gerrit/adapter/adapter.go

```diff
@@ -166,12 +174,39 @@ func (c *Controller) applyGlobalConfigOnce(cfg config.Getter, gerritClient *clie
 	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
 }
 
+// Helper function to create the cache used for InRepoConfig. Currently only attempts to create cache and returns nil if failed.
+func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, configAgent *config.Agent) (cache *config.InRepoConfigCache, err error) {
+	opts := git.ClientFactoryOpts{
+		CloneURI:       cloneURI.String(),
+		Host:           cloneURI.Host,
+		CookieFilePath: cookieFilePath,
+	}
+	gc, err := git.NewClientFactory(opts.Apply)
```

Comment:
err is non nil only when `ioutil.TempDir` failed, see https://github.com/kubernetes/test-infra/blob/5bc58aa2c9045da8549bf21a284f40d1160348c3/prow/git/v2/client_factory.go#L140. I think this is a legit error and it's totally fine to error out.

Reply:
Done